src/plugins/dashboard/public/services/custom_branding/custom_branding_service.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { KibanaPluginServiceFactory } from '@kbn/presentation-util-plugin/public';
+import { DashboardStartDependencies } from '../../plugin';
+import { DashboardCustomBrandingService } from './types';
+
+export type CustomBrandingServiceFactory = KibanaPluginServiceFactory<
+  DashboardCustomBrandingService,
+  DashboardStartDependencies
+>;
+
+export const customBrandingServiceFactory: CustomBrandingServiceFactory = ({ coreStart }) => {
+  const { customBranding } = coreStart;
+  return {
+    customBranding: {
```

Comment:
Should customBrandingServiceFactory return type be `{ hasCustomBranding$ }` so you don't have calls like `customBranding.customBranding.hasCustomBranding$` in [src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx](https://github.com/elastic/kibana/pull/149253/files?file-filters%5B%5D=.json&file-filters%5B%5D=.snap&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&owned-by%5B%5D=nreese&show-viewed-files=true#diff-05039dbe9dfc35fdd72d52d713e7732e2306098bceb50483d72c51ba5b7da721)?

Reply:
Good catch! [709d0b1](https://github.com/elastic/kibana/pull/149253/commits/709d0b1e1389a40b869fd40b393be246a2cb79f8)